File: scratch/convert_images.js
const fs = require('fs');
const path = require('path');
const sharp = require('sharp');

const brainDir = 'C:\\Users\\hp\\.gemini\\antigravity-ide\\brain\\3a1bdd74-6523-43cf-ae5f-39cb8b2a1fc4';
const destDir = path.join(__dirname, '..', 'public', 'images', 'sweets');

if (!fs.existsSync(destDir)) {
  fs.mkdirSync(destDir, { recursive: true });
}

const sweets = {
  motichoor_ladoo: 'motichoor_ladoo',
  kaju_katli: 'kaju_katli',
  gulab_jamun: 'gulab_jamun',
  rasgulla: 'rasgulla',
  peda: 'peda',
  milk_cake: 'milk_cake',
  barfi: 'barfi',
  jalebi: 'jalebi'
};

console.log('Starting Node.js image conversion...');

const files = fs.readdirSync(brainDir);

Object.entries(sweets).forEach(([sweetKey, destName]) => {
  // Find all files matching sweetKey_*.png
  const matches = files.filter(f => f.startsWith(sweetKey + '_') && f.endsWith('.png'));
  
  if (matches.length === 0) {
    console.log(`No match found for key: ${sweetKey}`);
    return;
  }
  
  // Sort by modification time to get the latest
  const sorted = matches.map(name => {
    const fullPath = path.join(brainDir, name);
    const stat = fs.statSync(fullPath);
    return { name, fullPath, mtime: stat.mtimeMs };
  }).sort((a, b) => b.mtime - a.mtime);
  
  const sourcePng = sorted[0].fullPath;
  const destWebp = path.join(destDir, `${destName}.webp`);
  
  console.log(`Converting ${path.basename(sourcePng)} -> ${destName}.webp`);
  
  sharp(sourcePng)
    .webp({ quality: 80 })
    .toFile(destWebp)
    .then(() => {
      console.log(`Successfully saved to ${destWebp}`);
    })
    .catch(err => {
      console.error(`Failed to convert ${sweetKey}:`, err);
    });
});

File: scratch/convert_images.py
import os
import glob
from PIL import Image # pyrefly: ignore [missing-import]

brain_dir = r"C:\Users\hp\.gemini\antigravity-ide\brain\3a1bdd74-6523-43cf-ae5f-39cb8b2a1fc4"
dest_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "public", "images", "sweets"))

os.makedirs(dest_dir, exist_ok=True)

sweets = {
    "motichoor_ladoo": "motichoor_ladoo",
    "kaju_katli": "kaju_katli",
    "gulab_jamun": "gulab_jamun",
    "rasgulla": "rasgulla",
    "peda": "peda",
    "milk_cake": "milk_cake",
    "barfi": "barfi",
    "jalebi": "jalebi"
}

print("Starting conversion...")
for sweet_key, dest_name in sweets.items():
    # Find files starting with sweet_key and ending with .png in the brain directory
    search_path = os.path.join(brain_dir, f"{sweet_key}_*.png")
    matches = glob.glob(search_path)
    if not matches:
        print(f"No match found for key: {sweet_key}")
        continue
    
    # Get the latest generated image matching this key
    matches.sort(key=os.path.getmtime)
    source_png = matches[-1]
    
    dest_webp = os.path.join(dest_dir, f"{dest_name}.webp")
    print(f"Converting {os.path.basename(source_png)} -> {dest_name}.webp")
    
    try:
        with Image.open(source_png) as img:
            img.save(dest_webp, "WEBP", quality=85)
        print("Success!")
    except Exception as e:
        print(f"Failed to convert: {e}")

print("Image conversion complete.")
